Firestore/core/src/firebase/firestore/model/precondition.h

```diff
@@ -54,6 +55,12 @@ class Precondition {
   /** Returns a precondition representing no precondition. */
   static Precondition None();
 
+  /**
+   * Default constructor which is equivalent to Precondition::None(). Prefer
+   * calling Precondition::None for readability.
```

Comment:
The real trouble is not to do with `Precondition` itself, but rather with types that embed it. Not default constructible is viral and I added this specifically because I added a usage to some other type that was prevented by this property. For little value types like this, there's more harm than good by making things not default constructible.